File: src/main/java/org/goafabric/calleeservice/service/CalleeService.java
package org.goafabric.calleeservice.service;

import org.goafabric.calleeservice.logic.CalleeLogic;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.MediaType;
import org.springframework.web.bind.annotation.*;

@RestController
@RequestMapping(value = "callees", produces = MediaType.APPLICATION_JSON_VALUE)
public class CalleeService {
    @Autowired
    CalleeLogic calleeLogic;

    @GetMapping("sayMyName")
    public Callee sayMyName (@RequestParam("name") String name) {
        return calleeLogic.sayMyName(name);
    }

    @GetMapping("sayMyOtherName/{name}")
    public Callee sayMyOtherName(@PathVariable("name") String name) {
        return calleeLogic.sayMyOtherName(name);
    }

    @PostMapping(value = "save", consumes = MediaType.APPLICATION_JSON_VALUE)
    public Callee save(@RequestBody Callee callee) {
        return calleeLogic.save(callee);
    }

}

File: src/main/java/org/goafabric/calleeservice/service/CalleeService.kt
package org.goafabric.calleeservice.service

import org.goafabric.calleeservice.logic.CalleeLogic
import org.springframework.http.MediaType.APPLICATION_JSON_VALUE
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RequestParam
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping(value = ["callees"], produces = [APPLICATION_JSON_VALUE])
class CalleeService (
    private val calleeLogic : CalleeLogic) {

    @GetMapping("sayMyName")
    fun sayMyName (@RequestParam("name") name : String) : Callee {
        return calleeLogic.sayMyName(name)
    }

    @GetMapping("sayMyOtherName/{name}")
    fun sayMyOtherName (@PathVariable("name") name : String) : Callee {
        return calleeLogic.sayMyOtherName(name)
    }

    @GetMapping("setSleepTime")
    fun setSleepTime (@RequestParam("sleepTime") sleepTime : Long) : Callee {
        return calleeLogic.setSleepTime(sleepTime)
    }
}